File: 2020-remove-one-element-to-make-the-array-strictly-increasing/2020-remove-one-element-to-make-the-array-strictly-increasing.java
class Solution {
    public boolean canBeIncreasing(int[] nums) {
        int prev = nums[0];
        boolean flag = false;
        for(int i=1;i<nums.length;i++){
            if(nums[i]<=prev){
                if(flag) return false;
                flag = true;
                if(i==1 || nums[i]>nums[i-2]){
                    prev = nums[i];
                }
            }else{
                prev = nums[i];
            }
        }
        return true;
    }
}

File: 2020-remove-one-element-to-make-the-array-strictly-increasing/2020-remove-one-element-to-make-the-array-strictly-increasing.cpp
class Solution {
public:
  bool check(vector<int>& arr){
        for(int i=1;i<arr.size();i++){
            if(arr[i-1]>=arr[i]) return false;
        }
        return true;
    }
    bool canBeIncreasing(vector<int>& nums) {
        for(int i=0;i<nums.size();i++){
            vector<int> arr;
            for(int j=0;j<nums.size();j++){
                if(i!=j){
                    arr.push_back(nums[j]);
                }
            }
            if(check(arr)) return true;

        }
        return false;
    }
};